app/src/main/java/com/example/garden_planner/EditGardenActivity.java

```diff
@@ -46,15 +69,218 @@ protected void onCreate(Bundle savedInstanceState) {
+
+                                    plants = new ArrayList<>();
+                                    deletionAdapter = new PlantDeletionAdapter(EditGardenActivity.this, plants);
+                                    GardenMethodHelper.queryPlantInBed(plants, deletionAdapter, garden);
+                                    rvCurrentPlants.setLayoutManager(new LinearLayoutManager(EditGardenActivity.this));
+                                    rvCurrentPlants.setAdapter(deletionAdapter);
+
+                                }
+                            });
+
+                            // create a planting reminder for plantInBed
+                            Reminder plantReminder = new Reminder();
+                            Date startPlantWeek = newPlant.getShouldPlantDate();
+                            Date endPlantWeek = GardenMethodHelper.convertToDate(
+                                    GardenMethodHelper.convertToLocalDateViaInstant(
+                                            startPlantWeek).plusWeeks(1));
+                            String reminderTitle = "Plant "+newPlantName;
+
+                            plantReminder.initializeReminder(startPlantWeek, endPlantWeek, reminderTitle,
+                                    selectedPlant.getPlantAdvice(), garden, newPlant, "plant");
+
+                            plantReminder.saveInBackground(new SaveCallback() {
+                                @Override
+                                public void done(ParseException e) {
+                                    if(e != null){
+                                        Log.e("yikes", e.getMessage());
+                                        Toast.makeText(EditGardenActivity.this, "Error in making the planting reminder!", Toast.LENGTH_SHORT).show();
+                                        return;
+                                    }
+
+                                }
+                            });
+                        }
+                    });
+
+                } catch (ParseException e) {
+                    e.printStackTrace();
+                }
+                // add search code
+            }
+        });
+
+        btSaveGarden.setOnClickListener(new View.OnClickListener() {
+            // display loading gifs as tasks finish
+            @Override
+            public void onClick(View v) {
+                List<PlantInBed> plantsToDelete = deletionAdapter.getPlantsToDelete();
+                for (PlantInBed plant : plantsToDelete){
+                    try {
+                        deleteReminders(plant);
+                        plant.delete();
+                    } catch (ParseException e) {
+                        e.printStackTrace();
+                    }
+                }
+
+                garden.saveInBackground(new SaveCallback() {
+                    // while loading, display the loading gif
+
+                    @Override
+                    public void done(ParseException e) {
```

Comment:
Do we need this ParseException e?

Reply:
It would be considered a different method if it didn't have it, so yes